client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -289,6 +302,26 @@ public StreamInfo getStreamInfo(String scopeName, String streamName) {
                                 });
     }
 
+    @Override
+    public <T> CompletableFuture<T> fetchEvent(EventPointer pointer, Serializer<T> serializer) {
+        Preconditions.checkNotNull(pointer);
+        Preconditions.checkNotNull(serializer);
+        CompletableFuture<T> completableFuture = CompletableFuture.supplyAsync(() -> {
+            @Cleanup
+            EventSegmentReader inputStream = inputStreamFactory.createEventReaderForSegment(pointer.asImpl().getSegment(), pointer.asImpl().getEventLength());
+            inputStream.setOffset(pointer.asImpl().getEventStartOffset());
```

Comment:
This pattern exists here and a few other places. It think it would be better to overload `createEventReaderForSegment` to have an optional offset, so that this can be done in one call. This already exists internally in the `getEventSegmentReader` call but isn't exposed.

Reply:
Yeah overloaded the createEventReaderForSegment in which we are passing all three segment, eventlength and start offset.